asgi_rabbitmq/core.py

```diff
@@ -960,102 +927,57 @@ def make_fernet(self, key):
         formatted_key = base64.urlsafe_b64encode(hashlib.sha256(key).digest())
         return Fernet(formatted_key)
 
-    def send(self, channel, message):
-        """Send the message to the channel."""
+    # API
 
+    async def send(self, channel, message):
+        """Send the message to the channel."""
         assert self.valid_channel_name(channel), 'Channel name is not valid'
-        future = self.thread.schedule(SEND, channel, message)
-        return future.result()
+        return await wrap_future(self.thread.schedule(SEND, channel, message))
 
-    def receive(self, channels, block=False):
+    async def receive(self, channel):
         """Receive one message from one of the channels."""
 
-        for channel in channels:
-            fail_msg = 'Channel name %s is not valid' % channel
-            assert self.valid_channel_name(channel, receive=True), fail_msg
-        future = self.thread.schedule(RECEIVE, channels, block)
-        try:
-            return future.result(timeout=self.receive_timeout)
-        except TimeoutError:
-            if not future.cancel():
-                return future.result()
-            return None, None
+        fail_msg = 'Channel name %s is not valid' % channel
+        assert self.valid_channel_name(channel, receive=True), fail_msg
+        return await wrap_future(self.thread.schedule(RECEIVE, [channel], True))
 
-    def new_channel(self, pattern):
+    async def new_channel(self, prefix="specific."):
         """Create new single reader channel."""
-
-        assert pattern.endswith('?')
+        # assert prefix.endswith('?')
         random_string = "".join(random.choice(ASCII) for i in range(20))
-        new_name = pattern + random_string
-        future = self.thread.schedule(NEW_CHANNEL, new_name)
-        future.result()
+        new_name = prefix + random_string
+        await wrap_future(self.thread.schedule(NEW_CHANNEL, new_name))
```

Comment:
looks like we create the channel here, but the official adapters only return a name